src/validator/MatchValidator.cpp

```diff
@@ -243,7 +243,7 @@ Status MatchValidator::buildEdgeInfo(const MatchPath *path,
                     return Status::SemanticError("`%s': Unknown edge type", type->c_str());
                 }
                 edgeInfos[i].edgeTypes.emplace_back(etype.value());
-                edgeInfos[i].types.emplace_back(*type);
+                edgeInfos[i].types.emplace_back(type.get());
```

Comment:
Is it possible for "type"(path->edge()->types()) to be released first here?

Reply:
The AST sentences is release when the query finished.